### DISCUSSION TITLE
Obtain a printable signature string

### DISCUSSION THREAD

**Discussion by Robert-vonZira** [tlsfuzzer/python-ecdsa#279] [Q&A] [answered]
ecdsaPrivateKey.sign_deterministic(b"message") 
returns a signature as a byte object - see keys.py: 

        :return: encoded signature for the `digest` hash
        :rtype: bytes or sigencode function dependant type

I want to get the signatures a string (to save it into txt file)

For privatekeys there is ta similar function: ecdsaPrivateKey.to_string().hex()) 
I can not find anything for signatures.

How could the signature data be transformed into a printable (and therefore txt file - writeable) string format?

Thank you in advance for your answer(s)!

**Comment by tomato42** [comment, marked as answer]
If you use sigencode function that returns `bytes()`, the calling `.hex()` on the returned object will give you a string:
```python
from ecdsa import SigningKey
sk = SigningKey.generate() # uses NIST192p
vk = sk.verifying_key
signature = sk.sign(b"message")
assert vk.verify(signature, b"message")
signature.hex()
```
will return something like
```
'4cb9aa58541fb1f6b3fca79e5d85a8e424545ccd8d82302622248f7ea47ad0e2e08c021dea99bec0005b3157ad5a17d3'
```
(the contents will vary, but that's the length you can expect for NIST192p)

**Reply by Robert-vonZira** [reply]
1. it works! THANK YOU!
2. but I tried hex() before... presumably I was to tired.
